app/models/miq_request_workflow.rb

```diff
@@ -47,24 +47,30 @@ def self.update_requester_from_parameters(data, user)
     new_user
   end
 
-  def initialize(values, requester, options = {})
-    instance_var_init(values, requester, options)
+  def initialize_dialogs(values, options)
+    return if options[:skip_dialog_load] == true
 
-    unless options[:skip_dialog_load] == true
-      # If this is the first time we are called the values hash will be empty
-      # Also skip if we are being called from a web-service
-      if @dialogs.nil?
-        @dialogs = get_dialogs
-        normalize_numeric_fields
-      else
-        @running_pre_dialog = true if options[:use_pre_dialog] != false
-      end
+    # If this is the first time we are called the values hash will be empty
+    # Also skip if we are being called from a web-service
+    if @dialogs.nil?
+      @dialogs = get_dialogs
+    else
+      @running_pre_dialog = true if options[:use_pre_dialog] != false
     end
+    normalize_numeric_fields unless @dialogs.nil?
+  end
 
-    unless options[:skip_dialog_load] == true
-      set_default_values
-      update_field_visibility
-    end
+  def configure_dialogs(values, options)
+    return if options[:skip_dialog_load] == true
```

Comment:
I wonder if it would be cleaner to have this conditional outside of the method and in the initialize.  That way a) you don't need options passed into here, and b) the method just has one job unconditionally.

Same goes for the initialize_dialogs method,

Reply:
Maybe - I could do that.  Also, wondering now that I re-ordered the initialize method [here](https://github.com/ManageIQ/manageiq/pull/21014/commits/77b9deacacf32697558e24b892cc0d72d9a091d5) whether we need to separate `initialize_dialogs` vs `configure_dialogs`